src/qml/QFieldCloudPopup.qml

```diff
@@ -413,6 +412,67 @@ Popup {
             onClicked: projectUpload(false)
           }
 
+          RowLayout {
+            Layout.leftMargin: 10
+            Layout.rightMargin: 10
+
+            Label {
+              Layout.fillWidth: true
+              Layout.alignment: Qt.AlignVCenter
+              topPadding: 10
+              bottomPadding: 10
+              font: Theme.tipFont
+              wrapMode: Text.WordWrap
+              color: autoPush.checked ? Theme.mainTextColor : Theme.secondaryTextColor
+
+              text:  qsTr('Automatically push changes every %n minute(s)','',0 + cloudProjectsModel.currentProjectData.AutoPushDelay)
+
+              MouseArea {
+                anchors.fill: parent
+                onClicked: cloudProjectsModel.projectSetAutoPushEnabled(cloudProjectsModel.currentProjectId, !autoPush.checked)
+              }
+            }
+
+            QfSwitch {
+              id: autoPush
+              Layout.preferredWidth: implicitContentWidth
+              Layout.alignment: Qt.AlignVCenter
+              width: implicitContentWidth
+              small: true
+
+              checked: !!cloudProjectsModel.currentProjectData.AutoPushEnabled
+              onClicked:  {
+                cloudProjectsModel.projectSetAutoPushEnabled(cloudProjectsModel.currentProjectId, checked)
+              }
+            }
+
+            Timer {
+              id: autoPushTimer
+              running: !!cloudProjectsModel.currentProjectData.AutoPushEnabled
+              interval: (!cloudProjectsModel.currentProjectData.AutoPushDelay - 0) * 60 * 1000
+              repeat: true
+
+              onRunningChanged: {
+                if (running && pushButton.enabled) {
+                  var dt = cloudProjectsModel.currentProjectData.LastLocalPushDeltas
+                  if (dt) {
+                    dt = new Date(dt)
+                    var now = new Date()
+                    if ((now - dt) >= interval) {
+                      projectUpload(false)
+                    }
+                  }
```

Comment:
```suggestion
                  const dtStr = cloudProjectsModel.currentProjectData.LastLocalPushDeltas

                  if (dtStr) {
                    const dt = new Date(dtStr)
                    const now = new Date()

                    if ((now - dt) >= interval) {
                      projectUpload(false)
                    }
                  }
```
I would avoid type conversions whenever possible, especially if I can use the `const` instead of `let`.